Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/address/AbstractAddressSpace.java

```diff
@@ -383,7 +383,7 @@ public Address addNoWrap(Address addr, long displacement) throws AddressOverflow
 				"Address Overflow in add: " + addr + " + 0x" + Long.toHexString(displacement));
 		}
 		long addrOff = addr.getOffset();
-		long maxOff = maxAddress.getOffset();
+		long maxOff = maxAddress.getOffset() + 1;
```

Comment:
This change will defeat address space wrap detection for 64bit addresses.

Reply:
So, should be +1L, or what's wrong?

Reply:
the +1 isn't failing, it the way the logic a few lines down fails.

A 64bit address space's max address is special-cased because we are storing the offset in a 64bit value.  If you +1 the max address, you've broken the 'hacks' in the conditionals that are looking for that.

Reply:
How can we fix that?

Reply:
Sure, we can move the addition in the comparement statement.

Reply:
I was probably being too cautious in smacking this suggested change down and only pointing out the issue with 64bit address spaces.  This method's raison d'être is to prevent the operation from exceeding the max address.

In your screenshot, you've got result=0x80040a9c and max=0x80040a9b.  The result has exceeded the max address and should fail.

The code that is calling this probably has faulty logic for dealing with the end of the address space.  Can you post a stack trace in #3840 of when this happens for the last address?

Reply:
What happens during calling of that function is to add to two last instructions sizes (one `jr ra` and one `_nop`) to the current address to check if it belongs to the current segment, but the `addNoWrap` throws the exception.

![image](https://user-images.githubusercontent.com/7189309/148837592-1068966e-7bae-41f3-8ff9-f1d8917e7573.png)

Reply:
If you need a test file, use `OVERLAY1.BIN` from the mentioned issue, and load it at `0x8004099C`.

Reply:
`addNoWrap()` compares wrong instances in result: max offset of a segment and next available address of it, that's it.

Reply:
If you will just set up breakpoint at that exception throwing, and will start a new program analyzer, it will break many and many times there during a normal program disassemblying.

Reply:
@dev747368   `if (result < addrOff || result >= maxOff) {` instead or would that have the same breakage?

Reply:
> @dev747368 `if (result < addrOff || result >= maxOff) {` instead or would that have the same breakage? (assuming this is the place to fix)

No. Look at the original issue to see values that are compared at this line. Your fix won't change anything, because it throws because the right part is lower.

Reply:
Closing this PR.  See fix in #3840.